source/MaterialXTest/ShaderValid.cpp

```diff
@@ -0,0 +1,291 @@
+#include <MaterialXTest/Catch/catch.hpp>
+
+#include <MaterialXCore/Document.h>
+#include <MaterialXFormat/XmlIo.h>
+
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/GlslShaderGenerator.h>
+#include <MaterialXShaderGen/Util.h>
+#include <MaterialXShaderGen/ShaderGenerators/Common/Swizzle.h>
+#include <MaterialXShaderGen/HwShader.h>
+
+#include <fstream>
+
+namespace mx = MaterialX;
+
+// Compile if module flags were set
+#if defined(MATERIALX_TEST_VIEW) && defined(MATERIALX_BUILD_VIEW)
+// Run only on windows for now
+#if defined(_WIN32)
+#include <iostream>
+#include <MaterialXView/ShaderValidators/Glsl/GlslValidator.h>
+#include <MaterialXView/Glsl/external/tinyexr/TinyEXRImageHandler.h>
+
+TEST_CASE("GLSL Validation from Source", "[view]")
```

Comment:
Not much differs between input from files on disk and hwShader. More work will be required for things not possible to parse from shader code alone.